### DISCUSSION TITLE
Filamentphp Import Action - add a select input field in the mapping form

### DISCUSSION THREAD

**Discussion by bladerunner2021** [filamentphp/filament#10431] [Help]
Hello,
Is there a chance to add to the import mapping form a  select input  to allow user to select for example a category or a tag to add to the import process. Thanks

**Comment by semno** [comment]
hi,
Just add the following function to the Importer class — that’s all you need:


```
public static function getOptionsFormComponents(): array
    {
        return [
            Select::make('category')
            ->relationship('category', 'name'),
        ];
    }
```

_This method (getOptionsFormComponents) is responsible for customizing the form attribute. So if you need to add or modify any fields in the future, this is the place to do it.

Everything else should work seamlessly._

reference => [https://filamentphp.com/docs/3.x/actions/prebuilt-actions/import#using-import-options](https://filamentphp.com/docs/3.x/actions/prebuilt-actions/import#using-import-options)

**Reply by zeeshantariq08** [reply]
@semno  how to show the select above the file selector?
